src/rlang/c-utils.h

```diff
@@ -23,6 +23,25 @@
 void* r_shelter_deref(r_obj* x);
 
 
+// Allow integers up to 2^52, same as R_XLEN_T_MAX when long vector
+// support is enabled
+#define RLANG_MAX_DOUBLE_INT 4503599627370496
+
+static inline
+bool r_dbl_is_decimal(double x) {
```

Comment:
I think this name is wrong? Is this `r_dbl_is_whole()`?

i.e. it returns `true` if the double value is "kind of like an integer"

`r_dbl_is_decimal()` implies that `r_dbl_is_decimal(1.5)` returns `true`, i think

Reply:
oops